### PR TITLE
Fixed grammatical errors in the description of settings in Settings.h

### PR DESCRIPTION
**Pull request by Baitur5**
### Changelog category (leave one):
- Documentation (changelog entry is not required)

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-clickhouse-ci-1** [comment]
*This is an automated comment for commit 8b7b3b249d3da87a6da6af3c135b91505a0afa6f with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/69764/8b7b3b249d3da87a6da6af3c135b91505a0afa6f/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/69717/e0fc95c894fcd129566b7c374496423e30f94c2c/docs_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

**Comment by alexkats** [comment]
Depends on #69213

**Review by alexkats — CHANGES_REQUESTED** [review]
In general it's ok, but there are conflicts that need to be resolved

File: src/Core/Settings.cpp [hunk 1]
```diff
@@ -357,11 +357,11 @@ namespace ErrorCodes
...
+    M(Int64, distributed_ddl_task_timeout, 180, "Timeout for DDL query responses from all hosts in the cluster. If a ddl request has not been performed on all hosts, a response will contain a timeout error and a request will be executed in an async mode. A negative value means infinite. Zero means async mode.", 0) \
     M(Milliseconds, stream_flush_interval_ms, 7500, "Timeout for flushing data from streaming storages.", 0) \
     M(Milliseconds, stream_poll_timeout_ms, 500, "Timeout for polling data from/to streaming storages.", 0) \
     \
-    M(Bool, final, false, "Query with the FINAL modifier by default. If the engine does not support final, it does not have any effect. On queries with multiple tables final is applied only on those that support it. It also works on distributed tables", 0) \
+    M(Bool, final, false, "Query with the FINAL modifier by default. If the engine does not support the final, it does not have any effect. On queries with multiple tables, FINAL is applied only to those that support it. It also works on distributed tables", 0) \
```
Comment (alexkats) [inline src/Core/Settings.cpp:364]:
```suggestion
    M(Bool, final, false, "Query with the FINAL modifier by default. If the engine does not support the FINAL, it does not have any effect. On queries with multiple tables, FINAL is applied only to those that support it. It also works on distributed tables", 0) \
```

**Review by alexkats — CHANGES_REQUESTED** [review]
Couple of minor things, other than that LGTM

File: src/Core/Settings.cpp [hunk 2]
```diff
@@ -472,7 +472,7 @@ namespace ErrorCodes
     M(UInt64, max_rows_in_join, 0, "Maximum size of the hash table for JOIN (in number of rows).", 0) \
     M(UInt64, max_bytes_in_join, 0, "Maximum size of the hash table for JOIN (in number of bytes in memory).", 0) \
     M(OverflowMode, join_overflow_mode, OverflowMode::THROW, "What to do when the limit is exceeded.", 0) \
-    M(Bool, join_any_take_last_row, false, "When disabled (default) ANY JOIN will take the first found row for a key. When enabled, it will take the last row seen if there are multiple rows for the same key. Can be applied only to hash join and storage join.", IMPORTANT) \
```
[the remark this answers is not in the document]
Comment (Baitur5) [reply src/Core/Settings.cpp]: I guess you are right

**Review by alexkats — APPROVED** [review]

**Review by alexkats — CHANGES_REQUESTED** [review]
Test failure is related, you should change tests/queries/0_stateless/01221_system_settings.reference to reflect the updated description for `send_timeout` setting

And since you're gonna make changes anyway, could you please squash commits to have a cleaner history

**Comment by Baitur5** [comment]
> Test failure is related, you should change tests/queries/0_stateless/01221_system_settings.reference to reflect the updated description for `send_timeout` setting
> 
> And since you're gonna make changes anyway, could you please squash commits to have a cleaner history



> Test failure is related, you should change tests/queries/0_stateless/01221_system_settings.reference to reflect the updated description for `send_timeout` setting
> 
> And since you're gonna make changes anyway, could you please squash commits to have a cleaner history

I didn't get why check  02532_send_logs_level_test failed,since it doesn't seem to have anything with the things I changed

**Comment by alexkats** [comment]
Stateless tests (tsan) [1/4] - #49156

**Review by alexkats — APPROVED** [review]

**Comment by alexkats** [comment]
Please, don't update this PR, I need to resolve conflicts in the cloud fork, then this PR can be merged

**Comment by Baitur5** [comment]
> Please, don't update this PR, I need to resolve conflicts in the cloud fork, then this PR can be merged

Ok

File: src/Core/Settings.cpp [hunk 3]
```diff
@@ -567,20 +567,20 @@ namespace ErrorCodes
...
     M(Bool, formatdatetime_parsedatetime_m_is_month_name, true, "Formatter '%M' in functions 'formatDateTime()' and 'parseDateTime()' print/parse the month name instead of minutes.", 0) \
     M(Bool, parsedatetime_parse_without_leading_zeros, true, "Formatters '%c', '%l' and '%k' in function 'parseDateTime()' parse months and hours without leading zeros.", 0) \
     M(Bool, formatdatetime_format_without_leading_zeros, false, "Formatters '%c', '%l' and '%k' in function 'formatDateTime()' print months and hours without leading zeros.", 0) \
     \
-    M(UInt64, max_partitions_per_insert_block, 100, "Limit maximum number of partitions in single INSERTed block. Zero means unlimited. Throw exception if the block contains too many partitions. This setting is a safety threshold, because using large number of partitions is a common misconception.", 0) \
+    M(UInt64, max_partitions_per_insert_block, 100, "Limit maximum number of partitions in the single INSERTed block. Zero means unlimited. Throw an exception if the block contains too many partitions. This setting is a safety threshold because using a large number of partitions is a common misconception.", 0) \
```
Comment (rschu1ze) [inline src/Core/Settings.cpp:578]: Thanks for adding `the` / `a`. In this particular case, I am not sure if it makes the grammar better or not :-)

Maybe as a follow up: Most setting descriptions are very specific and require background/implementation knowledge. What would be extremely helpful is if we go through each description, check what it really does, what its side-effect are, and try to rewrite the description in a way that a random database DBA without much prior knowledge of ClickHouse internals could make sense out of it.

Comment (Baitur5) [reply src/Core/Settings.cpp:578]: Thank you for the feedback! I’m still learning about ClickHouse internals, so there might be a few things I need to adjust.

But in this particular case, I added "the" to refer to a specific inserted block and "a" to indicate any large number of partitions. I might be wrong, but this is how I understood the description.

Comment (Baitur5) [reply src/Core/Settings.cpp:578]: > Thanks for adding `the` / `a`. In this particular case, I am not sure if it makes the grammar better or not :-)
> 
> Maybe as a follow up: Most setting descriptions are very specific and require background/implementation knowledge. What would be extremely helpful is if we go through each description, check what it really does, what its side-effect are, and try to rewrite the description in a way that a random database DBA without much prior knowledge of ClickHouse internals could make sense out of it.



Thank you for the feedback! I’m still learning about ClickHouse internals, so there might be a few things I need to adjust.

But in this particular case, I added "the" to refer to a specific inserted block and "a" to indicate any large number of partitions. I might be wrong, but this is how I understood the description.

**Pull request merged**